python/pyspark/sql/dataframe.py

```diff
@@ -351,8 +352,62 @@ def show(self, n=20, truncate=True, vertical=False):
         else:
             print(self._jdf.showString(n, int(truncate), vertical))
 
+    def _get_repl_config(self):
+        """Return the configs for eager evaluation each time when __repr__ or
+        _repr_html_ called by user or notebook.
+        """
+        eager_eval = self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.enabled", "false").lower() == "true"
+        console_row = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.maxNumRows", u"20"))
+        console_truncate = int(self.sql_ctx.getConf(
+            "spark.sql.repl.eagerEval.truncate", u"20"))
+        return (eager_eval, console_row, console_truncate)
```

Comment:
How about declaring those as `@property`?

Reply:
OK.

Reply:
Done in d4bf01a. Please check.